gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GroovyTranslator.java

```diff
@@ -181,24 +182,30 @@ protected String getSyntax(final TraversalOptionParent.Pick o) {
 
         @Override
         protected String getSyntax(final Number o) {
```

Comment:
if you go further down in this file you will find `LanguageTypeTranslator`  which is a `TypeTranslator` for `gremlin-language`. It's an extension of this one. I would think you need some changes there too so that we can generate inf/nan that matches the grammar.